packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -313,15 +318,37 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         return (minIdx, minVotes);
     }
 
-    // solhint-disable-next-line no-empty-blocks
     function resolveElection() external virtual override {
-        // TODO Check preconditions
-        // TODO Compare lists and -> fill TO_REMOVE, and TO_ADD
-        // TODO Move votes from TO_REMOVE to TO_ADD
-        // TODO Burn missing TO_REMOVE
-        // TODO Mint missing TO_ADD
-        // TODO Flip epochs
-        // TODO _electionStore().latestElectionDataIndex++;
+        ElectionStore storage store = _electionStore();
+        MemberToken memberToken = MemberToken(store.memberTokenAddress);
+        address[] memory nextEpochMembers = _currentElectionData().nextEpochMembers;
+        uint currentMembersSize = store.members.length;
+        uint nextEpochMembersSize = nextEpochMembers.length;
+
+        if (!isEpochFinished()) {
+            revert EpochNotFinished();
+        }
+
+        if (!isElectionEvaluated()) {
+            revert ElectionNotEvaluated();
+        }
+
+        for (uint i = 0; i < currentMembersSize; i++) {
```

Comment:
Back to burn all and mint again to simplify the logic (and reduce the chance of errors) to transfer the right token and prevent re-minting unburned tokens